### PR TITLE
Style bridge settings tab according to design

### PR DESCRIPTION
**Pull request by Half-Shot**
This PR styles the bridge settings tab to conform to the designs.

### No bridges

image

### Some bridges

image

### Show more / Show less

image

Outstanding issues:
- [x] Show more is shown all the time
- [x] Avatar is not grey when not available
- [x] Protocol icons are not specced yet, currently only network icons.

Design: https://www.figma.com/file/QYwqQz30lpAdXFrfrxxQkS/Bridges?node-id=1%3A303

### CODE REVIEW

**Comment by Half-Shot** [comment]
@nadonomy An issue came up about doing the "Show more/less" text. It's difficult to show the button *only* when the text is too long, because calculating that is usually done in the CSS layer. We could limit it by character count, but that could look weird when scaled.

The other problems were that links to the Workspace/Channel were not included in the design, so there isn't any way to make use of those links. That may be intentional, though we could tuck it under "Show more".

File: src/components/views/settings/BridgeTile.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            <div className="column-data">
+                <h3>{protocolName}</h3>
+                <p className="workspace-channel-details">
+                    {workspaceChannelDetails}
+                </p>
+                <p className={metadataClassname}>
+                    {creator} {bot}
+                </p>
+                <AccessibleButton kind="secondary" onClick={this._toggleVisible.bind(this)}>
+                    Show { this.state.visible ? "less" : "more" }
```
Comment (foldleft) [inline src/components/views/settings/BridgeTile.js]: i18n please

File: res/css/views/dialogs/_RoomSettingsDialogBridges.scss [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2020 New Vector Ltd.
```
Comment (turt2live) [inline res/css/views/dialogs/_RoomSettingsDialogBridges.scss]: foundation

File: src/components/views/settings/BridgeTile.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2020 New Vector Ltd
```
Comment (turt2live) [inline src/components/views/settings/BridgeTile.js]: foundation

File: src/components/views/settings/BridgeTile.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                name={ protocolName }
+                idName={ protocolName }
+                url={ avatarUrl }
+            />;
+        } else {
+            networkIcon = <div class="noProtocolIcon"></div>;
+        }
+
+
+        const workspaceChannelDetails = _t("Workspace: %(networkName)s      Channel: %(channelName)s", {
```
Comment (turt2live) [inline src/components/views/settings/BridgeTile.js]: not sure if the spacing will be persisted. Might be best to break this into two strings and CSS them

File: src/components/views/settings/BridgeTile.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+import PropTypes from 'prop-types';
+import {getHttpUriForMxc} from "matrix-js-sdk/src/content-repo";
+import {_t} from "../../../languageHandler";
+import {MatrixClientPeg} from "../../../MatrixClientPeg";
+import Pill from "../elements/Pill";
+import {makeUserPermalink} from "../../../utils/permalinks/Permalinks";
+import BaseAvatar from "../avatars/BaseAvatar";
+import AccessibleButton from "../elements/AccessibleButton";
+
+export default class BridgeTile extends React.PureComponent {
```
Comment (turt2live) [inline src/components/views/settings/BridgeTile.js]: `@replaceableComponent` please

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 6]
```diff
@@ -33,104 +30,16 @@ export default class BridgeSettingsTab extends React.Component {
         roomId: PropTypes.string.isRequired,
     };
 
+    constructor() {
+        super();
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: this doesn't seem to be doing anything

File: src/components/views/settings/tabs/room/BridgeSettingsTab.js [hunk 7]
```diff
@@ -151,14 +60,40 @@ export default class BridgeSettingsTab extends React.Component {
...
+                </ul>
+            </div>;
+        } else {
+            content = <p>{_t(
+                "This room isn’t bridging messages to any platforms. " +
+                "<a>Learn more.</a>", {},
+                {
+                    // TODO: We don't have this link yet: this will prevent the translators
+                    // having to re-translate the string when we do.
+                    a: sub => '',
```
Comment (turt2live) [inline src/components/views/settings/tabs/room/BridgeSettingsTab.js]: at least return `sub` so it doesn't get squashed out

File: src/components/views/settings/BridgeTile.js [hunk 8]
```diff
@@ -23,7 +23,8 @@ import Pill from "../elements/Pill";
 import {makeUserPermalink} from "../../../utils/permalinks/Permalinks";
 import BaseAvatar from "../avatars/BaseAvatar";
 import AccessibleButton from "../elements/AccessibleButton";
-
+import {replaceableComponent} from "../../../utils/replaceableComponent";
```
Comment (turt2live) [inline src/components/views/settings/BridgeTile.js]: somehow you've destroyed the blank line between this and the decorator - plz fix

Comment (Half-Shot) [reply src/components/views/settings/BridgeTile.js]: :boom:

Comment (Half-Shot) [reply src/components/views/settings/BridgeTile.js]: image

I suspect I mistook this for a blank line.

**Pull request merged**